File: WebView2/cs/JavaScriptMusicSample/NativeMediaPlayer/TrackMetadata.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

namespace NativeMediaPlayer
{
    /// <summary>
    /// This is a helper class to wrap the metadata for a single audio track in a way that it
    /// can be easily exposed to the JavaScript code.
    /// </summary>
    public sealed class TrackMetadata
    {
        /// <summary>
        /// Absolute URL to the audio file to play
        /// </summary>
        public string Src
        {
            get;
            set;
        } = string.Empty;

        /// <summary>
        /// The title of the track
        /// </summary>
        public string Title
        {
            get;
            set;
        } = string.Empty;

        /// <summary>
        /// The name of the artist for the track
        /// </summary>
        public string Artist
        {
            get;
            set;
        } = string.Empty;

        /// <summary>
        /// URL to an image that can be used as a thumbnail for the track (such as album art)
        /// </summary>
        public string ThumbnailSrc
        {
            get;
            set;
        } = string.Empty;

        public TrackMetadata(string src, string title, string artist, string thumbnailSrc)
        {
            Src = src;
            Title = title;
            Artist = artist;
            ThumbnailSrc = thumbnailSrc;
        }
    }
}

File: WebView2/cpp/JavaScriptMusicSample/NativeMediaPlayer/TrackMetadata.cpp
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

#include "pch.h"
#include "TrackMetadata.h"
#include "TrackMetadata.g.cpp"

namespace winrt::NativeMediaPlayer::implementation
{
    TrackMetadata::TrackMetadata(
        hstring const& src,
        hstring const& title,
        hstring const& artist,
        hstring const& thumbnailSrc) :
        src {src},
        title {title},
        artist {artist},
        thumbnailSrc {thumbnailSrc}
    { }
    hstring TrackMetadata::Src()
    {
        return src;
    }
    void TrackMetadata::Src(hstring const& value)
    {
        src = value;
    }
    hstring TrackMetadata::Title()
    {
        return title;
    }
    void TrackMetadata::Title(hstring const& value)
    {
        title = value;
    }
    hstring TrackMetadata::Artist()
    {
        return artist;
    }
    void TrackMetadata::Artist(hstring const& value)
    {
        artist = value;
    }
    hstring TrackMetadata::ThumbnailSrc()
    {
        return thumbnailSrc;
    }
    void TrackMetadata::ThumbnailSrc(hstring const& value)
    {
        thumbnailSrc = value;
    }
}
